repo-scripts/size-analysis/analysis-helper.ts

```diff
@@ -0,0 +1,779 @@
+ */
+function extractSymbolsFromNamedExportStatement(
+  node: ts.ExportDeclaration,
+  exportsFullList: MemberList
+): MemberList {
+  if (node.exportClause && ts.isNamedExports(node.exportClause)) {
+    const actualExports: string[] = [];
+    node.exportClause.elements.forEach(exportSpecifier => {
+      const reExportedSymbol: string = extractOriginalSymbolName(
+        exportSpecifier
+      );
+      // eg: export {foo as foo1 } from '...';
+      // if export is renamed, replace with new name
+      // reExportedSymbol: stores the original symbol name
+      // exportSpecifier.name: stores the renamed symbol name
+      if (isExportRenamed(exportSpecifier)) {
+        actualExports.push(exportSpecifier.name.escapedText.toString());
+        // reExportsMember stores all re-exported symbols in its orignal name. However, these re-exported symbols
+        // could be renamed by the re-export. We want to show the renamed name of the symbols in the final analysis report.
+        // Therefore, replaceAll simply replaces the original name of the symbol with the new name defined in re-export.
+        replaceAll(
+          exportsFullList,
+          reExportedSymbol,
+          exportSpecifier.name.escapedText.toString()
+        );
+      } else {
+        actualExports.push(reExportedSymbol);
+      }
+    });
+    // for named exports: requires a filter step which keeps only the symbols listed in the export statement.
+    filterAllBy(exportsFullList, actualExports);
+  }
+  return exportsFullList;
+}
+/**
+ * @param node compiler representation of a named export statement
+ * @param symbolLocation a map with module name as key and the resolved module location as value. (map is populated by parsing import statements)
+ * @param declarations a collection of symbols extracted from current file of context (either a dts file or a js file).
+ * This function exclusively handles named export statements that has no from clause, i.e: statements like export {LogLevel};
+ * first case : import then export
+ * example: import {a} from '...'; export {a}
+ * The function retrieves the location where the exported symbol is defined from the corresponding import statements.
+ *
+ * second case: declare first then export
+ * examples: declare const apps: Map<string, number>; export { apps };
+ * function foo(){} ; export {foo as bar};
+ * The function parses export clause of the statement and replaces symbol with its current name (if the symbol is renamed) from the declaration argument.
+ */
+function handleExportStatementsWithoutFromClause(
+  node: ts.ExportDeclaration,
+  symbolLocation: Map<string, string>,
+  declarations: MemberList
+): void {
+  if (node.exportClause && ts.isNamedExports(node.exportClause)) {
+    node.exportClause.elements.forEach(exportSpecifier => {
+      // export symbol could be renamed, we retrieve its current/renamed name
+      const exportedSymbolCurrentName = exportSpecifier.name.escapedText.toString();
```

Comment:
It sounds like you are using `myA` and `myB` in the following case, so that you won't match with the imports. Shouldn't we use the original name.
```
     import {a, b} from '...'
     export {a as myA, b as myB};
```

Further, we need to handle renamed imports as I mentioned above. We should be handle the following case

```
     import {a as AA, b as BB} from '...'
     export {AA as myA, BB as myB};
```